tango/integrations/transformers/__init__.py

```diff
@@ -5,6 +5,148 @@
+  or "transformers::AutoModelForCausalLM::from_config").
+
+  For example, to instantiate a pretrained transformer model from params:
+
+  .. testcode::
+
+      from tango.integrations.torch import Model
+
+      model = Model.from_params({
+          "type": "transformers::AutoModel::from_pretrained",
+          "pretrained_model_name_or_path": "epwalsh/bert-xsmall-dummy",
+      })
+
+  Or to instantiate a transformer model from params without loading pretrained weights:
+
+  .. testcode::
+
+      from tango.integrations.torch import Model
+
+      model = Model.from_params({
+          "type": "transformers::AutoModel::from_config",
+          "config": {"pretrained_model_name_or_path": "epwalsh/bert-xsmall-dummy"},
+      })
+
+  .. tip::
+
+        You can see a list of all of the available auto model constructors from transformers by running:
+
+        .. testcode::
+
+            from tango.integrations.torch import Model
+            from tango.integrations.transformers import *
+
+            for name in sorted(Model.list_available()):
+                if name.startswith("transformers::AutoModel"):
+                    print(name)
+
+        .. testoutput::
+            :options: +ELLIPSIS
+
+            transformers::AutoModel::from_config
+            transformers::AutoModel::from_pretrained
+            transformers::AutoModelForAudioClassification::from_config
+            transformers::AutoModelForAudioClassification::from_pretrained
+            ...
+
+- :class:`~tango.integrations.torch.Optimizer`: All optimizers from transformers are registered according
+  to their class names (e.g. "transformers::AdaFactor").
+
+  .. tip::
+
+        You can see a list of all of the available optimizers from transformers by running:
+
+        .. testcode::
+
+            from tango.integrations.torch import Optimizer
+            from tango.integrations.transformers import *
+
+            for name in sorted(Optimizer.list_available()):
+                if name.startswith("transformers::"):
+                    print(name)
+
+        .. testoutput::
+            :options: +ELLIPSIS
+
+            transformers::Adafactor
+            transformers::AdamW
+
+- :class:`~tango.integrations.torch.LRScheduler`: All learning rate scheduler function from transformers
+  are registered according to their type name (e.g. "transformers::linear").
+
+  .. tip::
+
+        You can see a list of all of the available scheduler functions from transformers by running:
+
+        .. testcode::
+
+            from tango.integrations.torch import LRScheduler
+            from tango.integrations.transformers import *
+
+            for name in sorted(LRScheduler.list_available()):
+                if name.startswith("transformers::"):
+                    print(name)
+
+        .. testoutput::
+            :options: +ELLIPSIS
+
+            transformers::constant
+            transformers::constant_with_warmup
+            transformers::cosine
```

Comment:
There should be a lot more: https://github.com/huggingface/transformers/blob/v4.15.0/src/transformers/optimization.py#L272

Reply:
There ARE a lot more. These are just the first 3, alphabetically. Notice the ellipsis on the next line.

Reply:
👍🏻